src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -46,12 +49,13 @@ public abstract class HybridCollectorManager implements CollectorManager<Collect
 
     private final int numHits;
     private final HitsThresholdChecker hitsThresholdChecker;
-    private final boolean isSingleShard;
     private final int trackTotalHitsUpTo;
     private final SortAndFormats sortAndFormats;
     @Nullable
     private final Weight filterWeight;
     private static final float boost_factor = 1f;
+    private final HybridQueryScoreDocsMerger<ScoreDoc> scoreDocsMerger = new HybridQueryScoreDocsMerger<>();
+    private final TopDocsMerger topDocsMerger = new TopDocsMerger(scoreDocsMerger);
```

Comment:
do we think we should init them via constructor? it can help us in unit testing by doing  dependency injection.

Reply:
That's the question Varun has. I think it's fine for now to have it like this, don't see a use case when we want to init those classes differently. It can be changed in future if needed.

Reply:
I've moved initialization to constructor and now topdocs merger is a constructor parameter, hope that's what you had in mind